lib/rules/named-grid-areas-no-invalid/index.js

```diff
@@ -0,0 +1,129 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const valueParser = require('postcss-value-parser');
+
+const ruleName = 'named-grid-areas-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	expectedToken: () => 'Expected cell token within string',
+	expectedSameNumber: () => 'Expected same number of cell tokens in each string',
+	expectedRectangle: (name) => `Expected single filled-in rectangle for "${name}"`,
+});
+
+function rule(actual) {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls('grid-template-areas', (decl) => {
```

Comment:
```suggestion
		root.walkDecls(/^grid-template-areas$/i, (decl) => {
```

Looking through the [commonly overlooked edge-cases](https://stylelint.io/developer-guide/rules#commonly-overlooked-edge-cases), we should account for uppercase properties.

Let's add one reject test for `a { GRID-TEMPLATE-AREAS: ""; }`